packages/flutter/lib/src/material/material.dart

```diff
@@ -548,76 +535,89 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
   }
 }
 
-class _RenderInkFeatures extends RenderProxyBox implements MaterialInkController {
-  _RenderInkFeatures({
-    RenderBox? child,
-    required this.vsync,
-    required this.absorbHitTest,
-    this.color,
-  }) : super(child);
+/// A [RenderBox] that paints [Splash]es.
+///
+/// Splash effects can be painted on an ancestor [Material] widget by obtaining
+/// an instance of this controller via [Material.of] and calling [addSplash].
+class SplashController extends RenderProxyBox {
+  /// Creates a splash controller.
+  SplashController({required this.vsync, this.color}) : super(null);
 
-  // This class should exist in a 1:1 relationship with a MaterialState object,
-  // since there's no current support for dynamically changing the ticker
-  // provider.
-  @override
+  /// The [TickerProvider] used by this controller.
+  ///
+  /// [Splash]es added to this controller with [addSplash] use this vsync
+  /// to drive their animations.
   final TickerProvider vsync;
 
-  // This is here to satisfy the MaterialInkController contract.
-  // The actual painting of this color is done by a Container in the
-  // MaterialState build method.
-  @override
+  /// Optionally specifies the color of the surface on which
+  /// splashes are being painted.
+  ///
+  /// The splash controller doesn't paint this color—the value is stored here,
+  /// and descendants can access it via `Material.of(context).color`.
+  /// Unlike typical [InheritedWidget] `.of(context)` methods, no updates
+  /// are sent when the color changes.
+  ///
+  /// If non-null, this [RenderBox] will absorb hit tests.
+  /// (The hit test behavior is usually inconsequential, since a splash
+  /// controller typically has an opaque parent, an opaque child, or both.)
```

Comment:
I find this part of doc a bit hard to understand, and I think explaining one point per paragraph makes it clearer. I also adjusted wording a bit to emphasize the nature of being optional. What do you think of the following version?
```
  /// Suggests a color for the surface where splashes are painted.
  ///
  /// This color does not directly determine the appearance of the splash; 
  /// it serves as a reference value that descendants can optionally use 
  /// via `Material.of(context).color`. 
  ///
  /// Unlike standard [InheritedWidget] patterns, changes to this 
  /// color will **not** trigger automatic updates through `of(context)`.
  ///
  /// Additionally, if this color is non-null, the associated [RenderBox] 
  /// will absorb hit tests. However, this behavior usually has minimal impact 
  /// since splash controllers are often nested within widgets that are 
  /// either fully opaque, or have opaque children.
```

Reply:
This looks _much_ better, thanks!